File: TodoJava/app/src/main/java/com/megait/todojava/Todo.java
package com.megait.todojava;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter @Setter @AllArgsConstructor @NoArgsConstructor @Builder
public class Todo {

    @PrimaryKey(autoGenerate = true)
    private int id;

    // To-do의 내용
    private String content;

    // 완료 여부
    private boolean completed;
}

File: TodoKotlin/app/src/main/java/com/example/todokotlin/Todo.kt
package com.example.todojava

import androidx.room.Entity
import androidx.room.PrimaryKey


@Entity

class Todo {
    @PrimaryKey(autoGenerate = true) //autoGenerate = 엔티티가 추가될때마다 번호를 자동으로 지정하겠다.
    var id: Int? = null

    //Todo 의 내용
    var content: String? = null

    //boolean타입은 사용못하지만 사용가능하게 해주는게 room
    var completed: Boolean? = null
}